br/pkg/storage/writer.go

```diff
@@ -149,6 +155,10 @@ func (u *bufferedWriter) Write(ctx context.Context, p []byte) (int, error) {
 				return bytesWritten, errors.Trace(err)
 			}
 			p = p[w:]
+			// continue buf because compressed data size may be less than Cap - Len
+			if u.buf.Compressed() {
```

Comment:
can you explain more how this `continue` fix the error ? seems it just  save once `u.uploadChunk`

Reply:
For normal files, writing n bytes means that `u.buf` will grow n bytes, so we can directly write chunkToFill bytes and u.buf can reach 5MB.
But for compressed files, writing n bytes means that `u.buf` will only grow m (m << n) bytes, now `u.buf` may still much less than 5MB, so we can upload here directly.